client/src/globalState.tsx

```diff
@@ -0,0 +1,285 @@
+import { ApolloError, useLazyQuery, useSubscription } from "@apollo/client";
+import React, { useContext, useEffect, useState } from "react";
+import { Item, User } from "../../shared/graphql/graphql";
+import { gqlGetPinboardByComposerId, gqlOnCreateItem } from "../gql";
+import { EXPAND_PINBOARD_QUERY_PARAM } from "./app";
+import type { PinboardData } from "./pinboard";
+import type { PayloadAndType } from "./types/PayloadAndType";
+import type { PerPinboard } from "./types/PerPinboard";
+import type { PreselectedPinboard } from "../../shared/graphql/extraTypes";
+import { isWorkflowStub } from "../../shared/graphql/extraTypes";
+
+interface GlobalStateContextShape {
+  userEmail: string;
+  userLookup: { [email: string]: User } | undefined;
+
+  activePinboardIds: string[];
+  activePinboards: PinboardData[];
+
+  payloadToBeSent: PayloadAndType | null;
+  clearPayloadToBeSent: () => void;
+
+  openPinboard: (pinboardData: PinboardData) => void;
+  closePinboard: (pinboardId: string) => void;
+  preselectedPinboard: PreselectedPinboard;
+  selectedPinboardId: string | null | undefined;
+  clearSelectedPinboard: () => void;
+
+  showNotification: (item: Item) => void;
+  hasWebPushSubscription: boolean | null | undefined;
+
+  errors: PerPinboard<ApolloError>;
+  hasError: boolean;
+  setError: (pinboardId: string, error: ApolloError | undefined) => void;
+  hasErrorOnOtherPinboard: (pinboardId: string) => boolean;
+
+  isExpanded: boolean;
+  setIsExpanded: (_: boolean) => void;
+
+  unreadFlags: PerPinboard<boolean>;
+  setUnreadFlag: (
+    pinboardId: string
+  ) => (unreadFlag: boolean | undefined) => void;
+  hasUnread: boolean;
+  hasUnreadOnOtherPinboard: (pinboardId: string) => boolean;
+
+  presetUnreadNotificationCount: number | undefined;
+}
+const GlobalStateContext = React.createContext<GlobalStateContextShape | null>(
+  null
+);
+
+const GlobalStateContextProvider = GlobalStateContext.Provider;
```

Comment:
super minor, but we don't really need to alias this now that its only being used in this component